client/executor/wasmi/src/lib.rs

```diff
@@ -155,30 +213,76 @@ impl<'a> Sandbox for FunctionExecutor<'a> {
 		val_ptr: Pointer<u8>,
 		val_len: WordSize,
 	) -> WResult<u32> {
-		let sandboxed_memory = self.sandbox_store.memory(memory_id).map_err(|e| e.to_string())?;
+		let sandboxed_memory = self
+			.inner
+			.sandbox_store
+			.borrow()
+			.memory(memory_id).map_err(|e| e.to_string())?;
+
+		match sandboxed_memory {
+			sandbox::Memory::Wasmi(sandboxed_memory) => {
+				match MemoryInstance::transfer(
+					&self.inner.memory,
+					val_ptr.into(),
+					&sandboxed_memory,
+					offset as usize,
+					val_len as usize,
+				) {
+					Ok(()) => Ok(sandbox_primitives::ERR_OK),
+					Err(_) => Ok(sandbox_primitives::ERR_OUT_OF_BOUNDS),
+				}
+			}
 
-		match MemoryInstance::transfer(
-			&self.memory,
-			val_ptr.into(),
-			&sandboxed_memory,
-			offset as usize,
-			val_len as usize,
-		) {
-			Ok(()) => Ok(sandbox_primitives::ERR_OK),
-			Err(_) => Ok(sandbox_primitives::ERR_OUT_OF_BOUNDS),
+			#[cfg(feature = "wasmer-sandbox")]
+			sandbox::Memory::Wasmer(sandboxed_memory) => {
+				let len = val_len as usize;
```

Comment:
These `memory_set` and `memory_get` are very similar. Looks like you can reduce copypaste with a generic code. Also I would like to know why you could not use `MemoryInstance::transfer` to access the memory.